domains/client/domain-operator/src/fetch_domain_bootstrap_info.rs

```diff
@@ -20,16 +21,22 @@ pub struct BootstrapResult<CBlock: BlockT> {
     pub imported_block_notification_stream: ImportNotifications<CBlock>,
 }
 
-pub async fn fetch_domain_bootstrap_info<Block, CBlock, CClient>(
+pub async fn fetch_domain_bootstrap_info<Block, CBlock, CClient, DomainBackend>(
     consensus_client: &CClient,
+    domain_backend: &DomainBackend,
     self_domain_id: DomainId,
 ) -> Result<BootstrapResult<CBlock>, Box<dyn std::error::Error>>
 where
     Block: BlockT,
     CBlock: BlockT,
     CClient: HeaderBackend<CBlock> + ProvideRuntimeApi<CBlock> + BlockchainEvents<CBlock>,
     CClient::Api: DomainsApi<CBlock, Block::Header>,
+    DomainBackend: Backend<Block>,
 {
+    // The genesis block is finalized when the chain is initialized, if `finalized_state`
+    // is non-empty meaning the domain chain is already started in the last run.
+    let is_domain_started = domain_backend.blockchain().info().finalized_state.is_some();
```

Comment:
Not sure I follow this.

When we set the raw_genesis, by default the genesis block is finalized irrespective of whether this is first run or a restart, no?

Reply:
ahhh I see okay 👍🏼